src/Service/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -0,0 +1,938 @@
+
+                if (_runtimeConfig.Entities.TryGetValue(entityName, out Entity? entity) && entity is not null)
+                {
+                    if (entity.GetRestEnabledOrPathSettings() is bool restEnabled)
+                    {
+                        if (!restEnabled)
+                        {
+                            continue;
+                        }
+                    }
+                }
+
+                SourceDefinition sourceDefinition = _metadataProvider.GetSourceDefinition(entityName);
+                HashSet<string> exposedColumnNames = GetExposedColumnNames(entityName, sourceDefinition.Columns.Keys.ToList());
+                HashSet<string> autoGeneratedPKColumnNames = new();
+
+                // Create component schema for FULL entity with all primary key columns (included auto-generated)
+                // which will typically represent the response body of a request or a stored procedure's request body.
+                schemas.Add(entityName, CreateComponentSchema(entityName, fields: exposedColumnNames));
+
+                if (dbObject.SourceType is not SourceType.StoredProcedure)
+                {
+                    // Create an entity's request body component schema excluding autogenerated primary keys.
+                    // A POST request requires any non-autogenerated primary key references to be in the request body.
+                    foreach (string primaryKeyColumn in sourceDefinition.PrimaryKey)
+                    {
+                        // Non-Autogenerated primary key(s) should appear in the request body.
+                        if (!sourceDefinition.Columns[primaryKeyColumn].IsAutoGenerated)
+                        {
+                            autoGeneratedPKColumnNames.Add(primaryKeyColumn);
+                            continue;
+                        }
+
+                        if (_metadataProvider.TryGetExposedColumnName(entityName, backingFieldName: primaryKeyColumn, out string? exposedColumnName)
+                            && exposedColumnName is not null)
+                        {
+                            exposedColumnNames.Remove(exposedColumnName);
+                        }
+                    }
+
+                    schemas.Add($"{entityName}_NoAutoPK", CreateComponentSchema(entityName, fields: exposedColumnNames));
+
+                    // Create an entity's request body component schema excluding all primary keys
+                    // by removing the tracked autogenerated primary key column names and removing them from
+                    // the exposedColumnNames collection.
+                    // The schema component without primary keys is used for PUT and PATCH operation request bodies because
+                    // those operations require all primary key references to be in the URI path, not the request body.
+                    foreach (string primaryKeyColumn in autoGeneratedPKColumnNames)
```

Comment:
At this point doesnt this hashset represented those PK that are NOT autogen?

Reply:
updated.